Update release docs with release category label process (#11099)
DataDog/integrations-core: docs/developer/process/agent-release/pre-release.md

Before each build is triggered:

> 1. Ensure all PRs have a `category/X` label. See list of Agent Release category labels [here](https://datadoghq.atlassian.net/wiki/spaces/agent/pages/2250408413/Merging+PRs+during+RC+development#PR-categories).

Reviewer: ```suggestion
1. Ensure all PRs have a `category/X` label. For more information, see the [list of Agent Release category labels](https://datadoghq.atlassian.net/wiki/spaces/agent/pages/2250408413/Merging+PRs+during+RC+development#PR-categories).
```

Reviewer 2: Also are we allowed to link internal sites like Atlassian on public docs?

Author: Good point, I'll leave the link out since the `category` labels can be seen on github. I'll add a link to that page to our internal Agent Releases confluence page.

> 2. Merge any fixes that have been approved, then pull `master`

Reviewer: ```suggestion
2. Merge any fixes that have been approved and pull `master`.
```

> 3. Release [all changed integrations](../integration-release.md#bulk-releases) with the exception of `datadog_checks_dev`

Reviewer: ```suggestion
3. Release [all changed integrations](../integration-release.md#bulk-releases) except for `datadog_checks_dev`.
```
